src/git-istage/Patching.cs

```diff
@@ -134,14 +133,19 @@ public static string ComputePatch(PatchDocument document, IEnumerable<int> lineI
 
         public static void ApplyPatch(string pathToGit, string workingDirectory, string patch, PatchDirection direction)
         {
-            var isUndo = direction == PatchDirection.Reset ||
-                         direction == PatchDirection.Unstage;
+            var isUndo = direction is PatchDirection.Reset or PatchDirection.Unstage;
             var patchFilePath = Path.GetTempFileName();
             var reverse = isUndo ? "--reverse" : string.Empty;
             var cached = direction == PatchDirection.Reset ? string.Empty : "--cached";
-            var arguments = $@"apply {cached} {reverse} --whitespace=nowarn ""{patchFilePath}""";
-
-            File.WriteAllText(patchFilePath, patch);
+            var patchLines = patch.Split('\n');
+            
+            // passing -v to git apply will output more useful information in case of a patch failure
+            var arguments = $@"apply -v {cached} {reverse} --whitespace=nowarn ""{patchFilePath}""";
+
+            // Write patch to file with Environment NewLine to ensure line endings are normalized
+            foreach (var pLine in patchLines)
+                File.AppendAllText(patchFilePath,  pLine + Environment.NewLine);
```

Comment:
Isn't that a longwinded way to say

```C#
File.WriteAllLines(patchFilePath, patchLines);
```

Reply:
Oops, My bad. I changed this to WriteAllLines.